docs/development.md

```diff
@@ -343,7 +343,17 @@ To run E2E locally, set `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, `AZURE_SUBSCRI
 ./scripts/ci-e2e.sh
 ```
 
-You can optionally set `AZURE_SSH_PUBLIC_KEY_FILE` to use your own SSH key.
+You can optionally set the following variables:
+
+| Variable                           | Description                                                                                                               | Default                                                                                         |
+|------------------------------------|---------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------|
+| `AZURE_SSH_PUBLIC_KEY_FILE`        | Use your own SSH key.                                                                                                     | `${GOPATH}/src/sigs.k8s.io/cluster-api-provider-azure/.sshkey`                                  |
+| `AZURE_CONTROL_PLANE_MACHINE_TYPE` | The machine type of the control plane.                                                                                    | `Standard_D2s_v3`                                                                               |
+| `AZURE_NODE_MACHINE_TYPE`          | The machine type of the node machine.                                                                                     | `Standard_D2s_v3`                                                                               |
```

Comment:
isn't machine type part of the general workload cluster customization (see #customizing-the-cluster-deployment above) and not really a test config option? Can the other customization variables like `AZURE_LOCATION` and `WORKER_MACHINE_COUNT` also be overriden?

Reply:
For CAPI environment variables like `WORKER_MACHINE_COUNT`, they can't be overridden and they are set programmatically like https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/638/files#diff-3d0a31130788192989e785058215450fR78. For CAPZ environment variables like `AZURE_LOCATION`, they are set in `./scripts/ci-e2e.sh`, which can be overridden. I will make a note in the docs. Also, I removed the use of  `${AZURE_RESOURCE_GROUP}` and `${AZURE_VNET_NAME}`, and directly use `${CLUSTER_NAME}` as the resource group name and  `${CLUSTER_NAME}-vnet` as the vnet name in `e2e/data/infrastructure-azure/cluster-template.yaml`. Please let me what you think of that.